packages/office-ui-fabric-react/src/components/Dialog/Dialog.base.tsx

```diff
@@ -82,12 +83,21 @@ export class DialogBase extends BaseComponent<IDialogProps, {}> {
       topButtonsProps,
       type,
       minWidth,
-      maxWidth
+      maxWidth,
+      modalProps
     } = this.props;
 
-    const modalProps = {
+    const mergedLayerProps: ILayerProps = {
+      ...(modalProps ? modalProps.layerProps : { onLayerDidMount })
+    };
+    if (onLayerDidMount && !mergedLayerProps.onLayerDidMount) {
+      mergedLayerProps.onLayerDidMount = onLayerDidMount;
+    }
+
+    const mergedModalProps = {
       ...DefaultModalProps,
-      ...this.props.modalProps
+      ...modalProps,
+      ...{ layerProps: mergedLayerProps }
```

Comment:
nit: Don't think the spread operator is necessary on `...{ layerProps: mergedLayerProps }`.